src/pcp.rs

```diff
@@ -179,11 +181,15 @@ pub enum PcpError {
 
 /// A value of a certain type. Implementations of this trait specify an arithmetic circuit that
 /// determines whether a given value is valid.
-pub trait Value<F, G>: Sized + PartialEq + Eq + Debug
+pub trait Value<F, G>:
+    Sized + PartialEq + Eq + Debug + TryFrom<(<Self as Value<F, G>>::Param, Vec<F>), Error = TypeError>
 where
     F: FieldElement,
     G: Gadget<F>,
 {
+    /// Parameters used to construct a value of this type from a vector of field elements.
+    type Param;
```

Comment:
I wish we had a better name for this than `Param`, though I'll admit I'm struggling to come up with one.